compiler/AST/AstPrintDocs.cpp

```diff
@@ -78,6 +78,12 @@ bool AstPrintDocs::enterFnSym(FnSymbol* node) {
 
 
 bool AstPrintDocs::enterModSym(ModuleSymbol* node) {
+  // If a module is not supposed to be documented, do not traverse into it (and
+  // skip the documentation).
+  if (node->hasFlag(FLAG_NO_DOC)) {
+      return false;
+  }
+
```

Comment:
I think it is. I view this more as blocking the traversal into the module, even though it also prevents the module documentation. A different ordering scheme might rely on the fact ModuleSymbol::printDocs() knows not to print itself, for example.